### ISSUE TITLE
mesa is probably having some serious problem

### ISSUE THREAD

**Issue by NeroReflex** [ChimeraOS/chimeraos#973]
Very rarely long compilation ends up in compiler crashing.
When relaunched it goes on fine. I have only seen this kind of memory corruption compiling mesa woth lto but since we don't this is strange.
Another problem pointing to mesa is that in 45 and 45_1 I was able to freely recompile mesa and switch version amongs 23.3.0 and 24. Now even attempting to rebuild mesa without valgring makes the os break completely.
This needs to be investigated: my ipothesis is that xorg-xwayland needs to be compiled after our custom mesa, and possibly some other package in common among our sessions too.

**Comment by 0sabbas0** [comment]
Very rarely long compilation ends up in compiler crashing.
> Do you have a stack trace?

**Comment by NeroReflex** [comment]
> Very rarely long compilation ends up in compiler crashing.
> 
> > Do you have a stack trace?

No, but I am compiling stuff just to see of it crashes. In such case I will post the printed stacktrace

**Comment by NeroReflex** [comment]
There we go. Just now.

```
fs/xfs/xfs_iomap.c:1467:1: internal compiler error: Errore di segmentazione
 1467 | }
      | ^
0x1f84ca6 internal_error(char const*, ...)
        ???:0
0x6f5c1c pop_scope()
        ???:0
0x6f6594 pop_file_scope()
        ???:0
0x7e4b9a c_common_parse_file()
        ???:0
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://gitlab.archlinux.org/archlinux/packaging/packages/gcc/-/issues> for instructions.
make[4]: *** [scripts/Makefile.build:244: fs/xfs/xfs_iomap.o] Error 1
make[3]: *** [scripts/Makefile.build:485: fs/xfs] Error 2
make[3]: *** Attesa per i processi non terminati....
```

**Comment by NeroReflex** [comment]
I haven't had any such problem again and I have no idea what causes that. I will close this and reopen if an issue appears again.
